src/lavinmq/amqp/stream_consumer.cr

```diff
@@ -37,6 +39,18 @@ module LavinMQ
         when Nil, Int, Time, "first", "next", "last"
         else raise LavinMQ::Error::PreconditionFailed.new("x-stream-offset must be an integer, a timestamp, 'first', 'next' or 'last'")
         end
+        case frame.arguments["x-stream-filter-value"]?
+        when String
+          @filter = frame.arguments["x-stream-filter-value"].to_s
```

Comment:
What about
```suggestion
          @filter = frame.arguments["x-stream-filter-value"].to_s.split(",")
```
to get rid of the `split` in `StreamQueueMessageStore#matching?`? That would get rid of some allocations for every call to `#matching?`.

Reply:
Yes, that's better 👍